docs/manual/working/javaGuide/main/templates/JavaTemplatesDependencyInjection.md

```diff
@@ -0,0 +1,41 @@
+# Dependency Injection with Templates
+
+Twirl templates can now be created with a constructor annotation using `@this`.  The constructor annotation means that Twirl templates can be injected into templates directly and can manage their own dependencies, rather than the controller having to manage dependencies not only for itself, but also for the templates it has to render.
+
+As an example, suppose a template has a dependency on a component `TemplateRenderingComponent`, which is not used by the controller.  
+
+First, add the `@Inject` annotation to Twirl in `build.sbt`:
+
+```scala
+TwirlKeys.constructorAnnotations += "@javax.inject.Inject()"
```

Comment:
This is set in [PlaySettings.scala](https://github.com/playframework/playframework/blob/2.6.0-M1/framework/src/sbt-plugin/src/main/scala/play/sbt/PlaySettings.scala#L57) already - is it really necessary for the user to set it in `build.sbt` as well?